execution_engine_testing/tests/src/test/regression/gh_1688.rs

```diff
@@ -0,0 +1,225 @@
+        .expect("should have contract");
+
+    assert!(
+        contract.named_keys().contains_key(NEW_KEY_NAME),
+        "expected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+    assert!(
+        !account.named_keys().contains_key(NEW_KEY_NAME),
+        "unexpected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+}
+
+#[ignore]
+#[test]
+fn should_run_gh_1688_regression_stored_contract_by_hash() {
+    let (mut builder, _contract_package_hash, contract_hash) = setup();
+
+    let put_key_request = {
+        let deploy = DeployItemBuilder::new()
+            .with_address(*DEFAULT_ACCOUNT_ADDR)
+            .with_stored_session_hash(contract_hash, METHOD_PUT_KEY, RuntimeArgs::default())
+            .with_empty_payment_bytes(runtime_args! { ARG_AMOUNT => *DEFAULT_PAYMENT, })
+            .with_authorization_keys(&[*DEFAULT_ACCOUNT_ADDR])
+            .with_deploy_hash([42; 32])
+            .build();
+
+        ExecuteRequestBuilder::new().push_deploy(deploy).build()
+    };
+
+    builder.exec(put_key_request).expect_success().commit();
+
+    let account = builder.get_account(*DEFAULT_ACCOUNT_ADDR).unwrap();
+    let contract = builder
+        .get_contract(contract_hash)
+        .expect("should have contract");
+
+    assert!(
+        contract.named_keys().contains_key(NEW_KEY_NAME),
+        "expected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+    assert!(
+        !account.named_keys().contains_key(NEW_KEY_NAME),
+        "unexpected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+}
+
+#[ignore]
+#[test]
+fn should_run_gh_1688_regression_stored_contract_by_name() {
+    let (mut builder, _contract_package_hash, contract_hash) = setup();
+
+    let put_key_request = {
+        let deploy = DeployItemBuilder::new()
+            .with_address(*DEFAULT_ACCOUNT_ADDR)
+            .with_stored_session_named_key(
+                CONTRACT_HASH_KEY,
+                METHOD_PUT_KEY,
+                RuntimeArgs::default(),
+            )
+            .with_empty_payment_bytes(runtime_args! { ARG_AMOUNT => *DEFAULT_PAYMENT, })
+            .with_authorization_keys(&[*DEFAULT_ACCOUNT_ADDR])
+            .with_deploy_hash([42; 32])
+            .build();
+
+        ExecuteRequestBuilder::new().push_deploy(deploy).build()
+    };
+
+    builder.exec(put_key_request).expect_success().commit();
+
+    let account = builder.get_account(*DEFAULT_ACCOUNT_ADDR).unwrap();
+    let contract = builder
+        .get_contract(contract_hash)
+        .expect("should have contract");
+
+    assert!(
+        contract.named_keys().contains_key(NEW_KEY_NAME),
+        "expected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
+    assert!(
+        !account.named_keys().contains_key(NEW_KEY_NAME),
+        "unexpected {} in {:?}",
+        NEW_KEY_NAME,
+        contract.named_keys()
+    );
```

Comment:
These lines seem to be duplicated across all tests – is that correct? If so, can we extract it?

Reply:
I agree, the difference seems to be in how the deploy is created, we could extract it out and build a certain deploy based on a flag?

Reply:
Fair point - did this in 89bf83740fc6694e1fade4ec3563fe7103a6226b